modules/openthread/platform/crypto_psa_key_allocator.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto_psa_key_allocator.h"
+
+#ifdef CONFIG_OPENTHREAD_PSA_NVM_BACKEND_KMU
+#include <cracen_psa_kmu.h>
+
+#define OPENTHREAD_KMU_SINGLE_KEY_SLOT_SIZE (2)
+/* Reserved for six 128-bit keys, and one 256-bit key */
+#define OPENTHREAD_KMU_DEDICATED_SLOTS	    (8)
+#endif /* CONFIG_OPENTHREAD_PSA_NVM_BACKEND_KMU */
+
+otError getKeyRef(otCryptoKeyRef *aInputKeyRef, psa_key_attributes_t *aAttributes)
+{
+	if (!aInputKeyRef) {
+		return OT_ERROR_INVALID_ARGS;
+	}
+
+#if defined(CONFIG_OPENTHREAD_PSA_NVM_BACKEND_ITS)
+	/* In this case we do not need make any adjustments to the input KeyRef or Attributes */
+	return OT_ERROR_NONE;
+
+#elif defined(CONFIG_OPENTHREAD_PSA_NVM_BACKEND_KMU)
+	/* Exit function if the key is outside the expected range dedicated to persistent lifetime.
+	 * Currently we supports only 7 keys defined in the openthread/src/core/crypto/storage.hpp
+	 * file. The first six keys are 128-bit length (1 KMU slot), and the last one is 256-bit
+	 * length (2 kmu slots). If the OPENTHREAD_KMU_DEDICATED_SLOTS is changed the new
+	 * formula for converting keys must be written, because we must take into account the shift
+	 * of two slots for ECDSA private key (7th key).
+	 */
+	if (*aInputKeyRef <= CONFIG_OPENTHREAD_PSA_ITS_NVM_OFFSET ||
+	    *aInputKeyRef > CONFIG_OPENTHREAD_PSA_ITS_NVM_OFFSET + OPENTHREAD_KMU_DEDICATED_SLOTS) {
+		return OT_ERROR_NONE;
```

Comment:
just wondering.. isn't it better to assert, if the ID is from the unexpected range? Would be easier in future, to capture such situation.

Then, maybe just check if ID is from 1-6 then allocate 1 slot, for ID=7, allocate 2 slots.

Reply:
If it is outside the range, it means that the key may be in the volatile space, so we should exit it with error NONE. I describe it better.

Reply:
Hmm, we don't assign slots amount here, just provide the starting point in KMU. I described it in the comment, and it is important in the future to move slots start of the new key (8th) by two slots, not 1 because of ECDSA key at 7th position. For now, it simplifies the code and is described.

Reply:
Yeah, there is no simple way to determine volatile and persistent keys, but I think there is nothing bad with this approach; we just skip volatile keys.